### ISSUE TITLE
Not receiving join-accept

### ISSUE THREAD

**Issue by WhiptongueHydra** [ropg/heltec_esp32_lora_v3#62]
Hi, sorry to bother you. Was just wondering what regional parameters you were using? I am setting up the node with the following params:
![image](https://github.com/user-attachments/assets/8510f940-017f-4775-a860-0bd7dfb2b359)
I have tried every type of regional parameters but the heltec still is unable to receive the join-accept or any downlinks:
![image](https://github.com/user-attachments/assets/3ff6c2af-1423-4b42-852c-f85f9b15dd64)
Just want to be able to rule this out as an issue, could not find which ones you were using in any of the documentation. Thanks :)

**Comment by WhiptongueHydra** [comment]
I am still having same issues unfortunately, I am not sure why. It will not receive downlinks.

**Comment by ropg** [comment]
Yeah, you probably/hopefully found [this issue](https://github.com/ropg/heltec_esp32_lora_v3/issues/40), which tells you everything I do, in the greatest level of detail, to get a working setup. If you're doing exactly the same thing and it doesn't work, I'm really not sure how to help further. Please do post if you figure it out.

**Comment by DadoKomar** [comment]
Hello Rop, hello guys.

I also run into this problem, I  am not a guru in LoRa protocol, but maybe found something. Argh, investigating already for 3 days. I was developing my LoRa device over Meteoscientific (has more settings and options) and everything was fine. I was joining still on first or second attempt. And I am getting from network still incremented Join Nonce:

RLB_PRO: LoRa: SF = 9, BW = 125.0 kHz, CR = 4/5, IQ: U
RLB_PRO: JoinRequest sent (DevNonce = 9) <-- Rx Delay start
RLB_PRO: 00000000: 00 00 00 00 00 00 00 00 00 08 00 00 00 00 00 d0  ................
RLB_PRO: 00000010: da 09 00 95 0b 91 da                             .......           
RLB_PRO: 
RLB_PRO: PHY:  Frequency = 868.300 MHz, TX = 16 dBm
RLB_PRO: LoRa: SF = 9, BW = 125.0 kHz, CR = 4/5, IQ: D
RLB_PRO: Opening Rx1 window (128 ms timeout)... <-- Rx Delay end 
RLB_PRO: Closing Rx1 window
RLB_PRO: JoinAccept (**JoinNonce = 39, previously 37**):   <<Here increase 2 as first receive was unsucesfull
RLB_PRO: 00000000: 20 27 00 00 3c 00 00 01 00 00 78 00 01 18 4f 84   '..<.....x...O.
RLB_PRO: 00000010: e8 56 84 b8 5e 84 88 66 84 58 6e 84 00 84 8e b5  .V..^..f.Xn.....
RLB_PRO: 00000020: e1                                               .                 
RLB_PRO: LoRaWAN revision: 1.0

Then I switched to Helium Console (much cheaper) and suddenly, device will join only very first time after flash erase and enter IDs. then It is hard to join, but after dozens attempts it will join. Not good. I found out Helium is sending some random Nonces (not increments), so maybe then radiolib is not accepting it? 

RLB_PRO: JoinRequest sent (DevNonce = 6) <-- Rx Delay start
RLB_PRO: 00000000: 00 00 00 00 00 00 00 00 00 09 00 00 00 00 00 d0  ................
RLB_PRO: 00000010: da 06 00 97 b9 26 b8                             .....&.           
RLB_PRO: 
RLB_PRO: PHY:  Frequency = 868.500 MHz, TX = 16 dBm
RLB_PRO: LoRa: SF = 9, BW = 125.0 kHz, CR = 4/5, IQ: D
RLB_PRO: Opening Rx1 window (128 ms timeout)... <-- Rx Delay end 
RLB_PRO: Closing Rx1 window
RLB_PRO: JoinAccept (**JoinNonce = 8040173, previously 15934174**):
RLB_PRO: 00000000: 20 ed ae 7a 48 65 32 6d 02 00 48 00 01 e4 37 bc   ..zHe2m..H...7.
RLB_PRO: 00000010: 33                                               3                 
RLB_PRO: [persist] Nonces and session data saved to RTC RAM
...
RLB_PRO: JoinAccept (**JoinNonce = 1434403, previously 15934174**):
...
RLB_PRO: JoinAccept (**JoinNonce = 398148, previously 15934174**):
...
RLB_PRO: JoinAccept (**JoinNonce = 12990835, previously 15934174**):


No problem with RX1 reception window, but that smaller new generated JoinNonce could be a problem. 
In Helium Console there are almost no settings. Dont know even where to clear nonces or where to set LoRa MAC version. I found only setting over profile: ADR support and RX1 window delay. 

Is this some Helium Console Nonce generation issue? Be carefull about this as when you leave joining it for long time (and small repeat OTAA time) it is eating DCs as hell. Each OTAA attempt 10 to 20 DC as I have 20 Access points near. 

**Rop, could you please input your thoughts? I will appreciate. 
Anyway great job with your library! It is reliable as should be and can run for months.**

Guys any input appreciated. Does somebody know where to set parameters in Helium console?
Does it mean Radiolib needs to be modyfied to accept also smaller new generated JoinNounce? Have some older Radiolib v7.1.2

**Comment by DadoKomar** [comment]
Hello Rop. So been digging into it for long. 

I realized this issue belongs under v1.0.x OTAA issues for Helium Legacy network. Joining validation is done still according v1.1 so is failing. First time fail it can be fine, but I guess then it should get from the network revision version and set  this->rev = 0 . It is still trying to validate according v1.1 for higher Nonce as was previous and then refusing to join.

Maybe if possible to enter NULL as nwkKey in your prompt, then RadioLib could treat it from start as v1.0.x  (nwkKey is not defined in v1.0.x and I see some validation for nwkKey pointer in BeginOTAA and then set  this->rev = 1    which is v1.1)  

We need to set to 0 if joining to v1.0.x networks.  

I was unable to set or find somewhere default version setting to 1.0.x  - I am tyred now. It is crazy how complex is that LoRa.
